src/config/ch.sbb.direktverbindungen/DvFeatureInfo/DvFeatureInfo.js

```diff
@@ -0,0 +1,278 @@
+  const classes = useStyles({ teaser });
+  const teaserOnClick = useCallback(() => {
+    return teaser ? setTeaser(false) : undefined;
+  }, [teaser]);
+  const [revision, forceRender] = useState();
+
+  const getVisibleLayerKeys = useCallback(
+    () =>
+      layers
+        .filter((l) => DV_DAY_NIGHT_REGEX.test(l.key) && l.visible)
+        .map((l) => l.key),
+    [layers],
+  );
+
+  const dvMainLayer = useMemo(
+    () => layers.find((l) => l.key === `${DV_KEY}.main`),
+    [layers],
+  );
+  const [layersVisible, setLayersVisible] = useState(getVisibleLayerKeys());
+
+  const dvFeatures = useMemo(() => {
+    const features = featureInfo.reduce((feats, info) => {
+      info.features.forEach((feat) => feat.set('layer', info.layer));
+      return [...feats, ...info.features];
+    }, []);
+    features.sort((feat) => (feat.get('line') === 'night' ? -1 : 1));
+    const cleaned = removeDuplicates(parseDvFeatures(features)).filter(
+      (feat) => !!feat.get('mapboxFeature'),
+    );
+    return filterByType
+      ? cleaned.filter(
+          (feat) =>
+            !!layersVisible.find((layerKey) => {
+              return `${DV_KEY}.${feat.get('line')}` === layerKey;
+            }),
+        )
+      : cleaned;
+  }, [featureInfo, layersVisible, filterByType]);
+
+  const previousFeatureInfo = usePrevious(featureInfo);
+  const previousDvFeatures = usePrevious(dvFeatures);
+  const previousInfoKey = usePrevious(infoKey);
+
+  useEffect(() => {
+    if (featureInfo !== previousFeatureInfo) {
+      setTeaser(true);
+      setInfoKey(getId(dvFeatures[0]));
+      return;
+    }
+    if (dvFeatures !== previousDvFeatures) {
+      const previousSelectedFeature = dvFeatures.find(
+        (feat) => getId(feat) === previousInfoKey,
+      );
+      if (previousInfoKey) {
+        setInfoKey(previousSelectedFeature && getId(previousSelectedFeature));
+      }
+    }
+  }, [
+    dvFeatures,
+    previousDvFeatures,
+    previousInfoKey,
+    layersVisible,
+    featureInfo,
+    previousFeatureInfo,
+  ]);
+
+  useEffect(() => {
+    const olKeys =
+      layers?.map((layer) => {
+        return layer?.on('change:visible', (evt) => {
+          forceRender(revision + 1);
+          const { target: targetLayer } = evt;
+          if (DV_DAY_NIGHT_REGEX.test(targetLayer.key)) {
+            setLayersVisible(getVisibleLayerKeys());
+            setInfoKey(null);
+          }
+        });
+      }) || [];
+    // Force render after first render because visibility of layers is  not yet applied.
+    if (revision === undefined) {
+      forceRender(0);
+      setLayersVisible(getVisibleLayerKeys());
+    }
+    return () => {
+      unByKey(olKeys);
+    };
+  }, [getVisibleLayerKeys, layers, layersVisible, revision]);
+
+  if (!dvFeatures?.length) {
+    return null;
+  }
+
+  return (
+    <>
+      {dvFeatures?.length ? (
+        <div className={classes.featureInfos}>
+          {dvFeatures.length > 1 ? (
```

Comment:
Can't we use the MenuItem (always opened)  when there is only one ?

Reply:
because it makes the code more readable, we'd need to make some micro adjustments to the MenuItem (prevent the teaser functionality, hide the toggle arrow etc) and avoids unnecessary HTML rendering (all the nested divs for collapsing etc. I think it's cleaner this way.